{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "f38d7db0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1109\n",
      "['A2ML1', 'AADAC', 'ABCA12', 'ABCA8', 'ABCA9', 'ABCB11', 'ABCB5', 'ABCG2', 'ABI3BP', 'ACAN']\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "\n",
    "df_DEG = pd.read_csv(r'string_data/string_DEG_S2.tsv',delimiter='\\t')\n",
    "\n",
    "arr_df_DEG_gene = []\n",
    "for i in df_DEG['#node1'].unique():\n",
    "    arr_df_DEG_gene.append(i)\n",
    "print(len(arr_df_DEG_gene))\n",
    "print(arr_df_DEG_gene[:10])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "5694949b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create a function to solve the repeated elements in the list：\n",
    "def kill_pardon(list_nokill):\n",
    "    f_list = []\n",
    "    for item in list_nokill:\n",
    "        if item not in f_list:\n",
    "            f_list.append(item)\n",
    "    return f_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "a0c70553",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1301\n",
      "['A2ML1', 'AADAC', 'ABCA12', 'ABCA8', 'ABCA9', 'ABCB11', 'ABCB5', 'ABCG2', 'ABI3BP', 'ACAN']\n"
     ]
    }
   ],
   "source": [
    "for i in df_DEG.node2.unique():\n",
    "    arr_df_DEG_gene.append(i)\n",
    "arr_df_DEG_gene = kill_pardon(arr_df_DEG_gene)\n",
    "print(len(arr_df_DEG_gene))\n",
    "print(arr_df_DEG_gene[:10])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "54433d10",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1748\n"
     ]
    }
   ],
   "source": [
    "# read from file\n",
    "df1 = pd.read_csv(r'string_data/DElnc-gene_S2.csv')\n",
    "df2 = pd.read_csv(r'string_data/DEmir-gene_S2.csv')\n",
    "df3 = pd.read_csv(r'string_data/DElnc-DEmiR_S2.csv') # ,keep_default_na=False, na_values=['_NA'])\n",
    "\n",
    "#df1.lncRNA.unique()\n",
    "arr_df1_lncRNA = []\n",
    "for i in df1.lncRNA.unique():\n",
    "    arr_df1_lncRNA.append(i)\n",
    "    \n",
    "#df1.gene.unique()\n",
    "arr_df1_gene = []\n",
    "for i in df1.gene.unique():\n",
    "    arr_df1_gene.append(i)\n",
    "\n",
    "#df2.miRNA.unique()\n",
    "arr_df2_miRNA = []\n",
    "for i in df2.miRNA.unique():\n",
    "    arr_df2_miRNA.append(i)\n",
    "    \n",
    "#df2.gene.unique()\n",
    "arr_df2_gene = []\n",
    "for i in df2.target_symbol.unique():\n",
    "    arr_df2_gene.append(i)\n",
    "    \n",
    "#df3.miRNA.unique()\n",
    "arr_df3_miRNA = []\n",
    "for i in df3.miR_DGE.unique():\n",
    "    arr_df3_miRNA.append(i)\n",
    "    \n",
    "#df3.lncRNA.unique()\n",
    "arr_df3_lncRNA = []\n",
    "for i in df3.geneName.unique():\n",
    "    arr_df3_lncRNA.append(i)\n",
    "\n",
    "arr_lncRNA = []\n",
    "arr_miRNA = []\n",
    "arr_gene = []\n",
    "\n",
    "for item in arr_df3_lncRNA:\n",
    "    arr_df1_lncRNA.append(item)\n",
    "for item in arr_df3_miRNA:\n",
    "    arr_df2_miRNA.append(item)\n",
    "for item in arr_df2_gene:\n",
    "    arr_df1_gene.append(item)\n",
    "for item in arr_df_DEG_gene:\n",
    "    arr_df1_gene.append(item)\n",
    "\n",
    "# select the genes in df_DEG：\n",
    "for item in arr_df1_gene:\n",
    "    if item in arr_df_DEG_gene:\n",
    "        arr_gene.append(item)\n",
    "\n",
    "arr_lncRNA = kill_pardon(arr_df1_lncRNA)\n",
    "arr_miRNA = kill_pardon(arr_df2_miRNA)\n",
    "arr_gene = kill_pardon(arr_gene)\n",
    "\n",
    "arr_matrix_columns = []\n",
    "for elem in arr_miRNA:\n",
    "    arr_matrix_columns.append(elem)\n",
    "for elem in arr_lncRNA:\n",
    "    arr_matrix_columns.append(elem)\n",
    "for elem in arr_gene:\n",
    "    arr_matrix_columns.append(elem)\n",
    "print(len(arr_matrix_columns))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "id": "301e2611",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>lncRNA</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABALON</td>\n",
       "      <td>AUTS2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABALON</td>\n",
       "      <td>CASP7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABALON</td>\n",
       "      <td>EBNA-3B/EBNA-3C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABALON</td>\n",
       "      <td>RBCK1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ABALON</td>\n",
       "      <td>ZNF706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11750</th>\n",
       "      <td>ZNF710-AS1</td>\n",
       "      <td>TBL1X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11751</th>\n",
       "      <td>ZNF710-AS1</td>\n",
       "      <td>KDM1B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11752</th>\n",
       "      <td>ZNF710-AS1</td>\n",
       "      <td>EGLN2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11753</th>\n",
       "      <td>ZNF710-AS1</td>\n",
       "      <td>ZNF598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11754</th>\n",
       "      <td>ZNF710-AS1</td>\n",
       "      <td>EBNA-2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11755 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           lncRNA             gene\n",
       "0          ABALON            AUTS2\n",
       "1          ABALON            CASP7\n",
       "2          ABALON  EBNA-3B/EBNA-3C\n",
       "3          ABALON            RBCK1\n",
       "4          ABALON           ZNF706\n",
       "...           ...              ...\n",
       "11750  ZNF710-AS1            TBL1X\n",
       "11751  ZNF710-AS1            KDM1B\n",
       "11752  ZNF710-AS1            EGLN2\n",
       "11753  ZNF710-AS1           ZNF598\n",
       "11754  ZNF710-AS1           EBNA-2\n",
       "\n",
       "[11755 rows x 2 columns]"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "id": "7796c1c2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>miRNA</th>\n",
       "      <th>target_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hsa-let-7b-5p</td>\n",
       "      <td>MT-ND4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hsa-let-7b-5p</td>\n",
       "      <td>SMARCC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hsa-let-7b-5p</td>\n",
       "      <td>TSC22D2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hsa-let-7b-5p</td>\n",
       "      <td>ATP6V1G1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hsa-let-7b-5p</td>\n",
       "      <td>SMARCB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13857</th>\n",
       "      <td>hsa-miR-98-5p</td>\n",
       "      <td>SMARCC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13858</th>\n",
       "      <td>hsa-miR-98-5p</td>\n",
       "      <td>CLUH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13859</th>\n",
       "      <td>hsa-miR-98-5p</td>\n",
       "      <td>CLINT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13860</th>\n",
       "      <td>hsa-miR-98-5p</td>\n",
       "      <td>PHACTR4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13861</th>\n",
       "      <td>hsa-miR-98-5p</td>\n",
       "      <td>PHKA1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>13862 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               miRNA target_symbol\n",
       "0      hsa-let-7b-5p        MT-ND4\n",
       "1      hsa-let-7b-5p       SMARCC1\n",
       "2      hsa-let-7b-5p       TSC22D2\n",
       "3      hsa-let-7b-5p      ATP6V1G1\n",
       "4      hsa-let-7b-5p       SMARCB1\n",
       "...              ...           ...\n",
       "13857  hsa-miR-98-5p       SMARCC1\n",
       "13858  hsa-miR-98-5p          CLUH\n",
       "13859  hsa-miR-98-5p        CLINT1\n",
       "13860  hsa-miR-98-5p       PHACTR4\n",
       "13861  hsa-miR-98-5p         PHKA1\n",
       "\n",
       "[13862 rows x 2 columns]"
      ]
     },
     "execution_count": 116,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "id": "fc3f7f45",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>miR_DGE</th>\n",
       "      <th>lnc_DGE</th>\n",
       "      <th>geneName</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hsa-let-7b-5p</td>\n",
       "      <td>ENSG00000251562</td>\n",
       "      <td>MALAT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hsa-let-7b-5p</td>\n",
       "      <td>ENSG00000250742</td>\n",
       "      <td>RP11-834C11.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hsa-let-7b-5p</td>\n",
       "      <td>ENSG00000203709</td>\n",
       "      <td>C1orf132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hsa-let-7b-5p</td>\n",
       "      <td>ENSG00000245910</td>\n",
       "      <td>SNHG6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hsa-let-7b-5p</td>\n",
       "      <td>ENSG00000203709</td>\n",
       "      <td>C1orf132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1772</th>\n",
       "      <td>hsa-miR-96-5p</td>\n",
       "      <td>ENSG00000234741</td>\n",
       "      <td>GAS5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1773</th>\n",
       "      <td>hsa-miR-96-5p</td>\n",
       "      <td>ENSG00000234741</td>\n",
       "      <td>GAS5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1774</th>\n",
       "      <td>hsa-miR-96-5p</td>\n",
       "      <td>ENSG00000234741</td>\n",
       "      <td>GAS5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775</th>\n",
       "      <td>hsa-miR-96-5p</td>\n",
       "      <td>ENSG00000234741</td>\n",
       "      <td>GAS5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1776</th>\n",
       "      <td>hsa-miR-98-5p</td>\n",
       "      <td>ENSG00000260244</td>\n",
       "      <td>RP11-588K22.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1777 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            miR_DGE          lnc_DGE       geneName\n",
       "0     hsa-let-7b-5p  ENSG00000251562         MALAT1\n",
       "1     hsa-let-7b-5p  ENSG00000250742  RP11-834C11.4\n",
       "2     hsa-let-7b-5p  ENSG00000203709       C1orf132\n",
       "3     hsa-let-7b-5p  ENSG00000245910          SNHG6\n",
       "4     hsa-let-7b-5p  ENSG00000203709       C1orf132\n",
       "...             ...              ...            ...\n",
       "1772  hsa-miR-96-5p  ENSG00000234741           GAS5\n",
       "1773  hsa-miR-96-5p  ENSG00000234741           GAS5\n",
       "1774  hsa-miR-96-5p  ENSG00000234741           GAS5\n",
       "1775  hsa-miR-96-5p  ENSG00000234741           GAS5\n",
       "1776  hsa-miR-98-5p  ENSG00000260244  RP11-588K22.2\n",
       "\n",
       "[1777 rows x 3 columns]"
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "1c1ec72d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#node1</th>\n",
       "      <th>node2</th>\n",
       "      <th>node1_string_id</th>\n",
       "      <th>node2_string_id</th>\n",
       "      <th>neighborhood_on_chromosome</th>\n",
       "      <th>gene_fusion</th>\n",
       "      <th>phylogenetic_cooccurrence</th>\n",
       "      <th>homology</th>\n",
       "      <th>coexpression</th>\n",
       "      <th>experimentally_determined_interaction</th>\n",
       "      <th>database_annotated</th>\n",
       "      <th>automated_textmining</th>\n",
       "      <th>combined_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A2ML1</td>\n",
       "      <td>CWH43</td>\n",
       "      <td>9606.ENSP00000299698</td>\n",
       "      <td>9606.ENSP00000226432</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.475</td>\n",
       "      <td>0.475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2ML1</td>\n",
       "      <td>DSG3</td>\n",
       "      <td>9606.ENSP00000299698</td>\n",
       "      <td>9606.ENSP00000257189</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.098</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.370</td>\n",
       "      <td>0.408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2ML1</td>\n",
       "      <td>PADI2</td>\n",
       "      <td>9606.ENSP00000299698</td>\n",
       "      <td>9606.ENSP00000364635</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.466</td>\n",
       "      <td>0.466</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2ML1</td>\n",
       "      <td>SERPINB7</td>\n",
       "      <td>9606.ENSP00000299698</td>\n",
       "      <td>9606.ENSP00000381101</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.550</td>\n",
       "      <td>0.550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A2ML1</td>\n",
       "      <td>KLK6</td>\n",
       "      <td>9606.ENSP00000299698</td>\n",
       "      <td>9606.ENSP00000366047</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.050</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.717</td>\n",
       "      <td>0.719</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8609</th>\n",
       "      <td>WNT2</td>\n",
       "      <td>WT1</td>\n",
       "      <td>9606.ENSP00000265441</td>\n",
       "      <td>9606.ENSP00000331327</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.084</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.385</td>\n",
       "      <td>0.413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8610</th>\n",
       "      <td>WNT2</td>\n",
       "      <td>WNT7B</td>\n",
       "      <td>9606.ENSP00000265441</td>\n",
       "      <td>9606.ENSP00000341032</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.925</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.668</td>\n",
       "      <td>0.904</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8611</th>\n",
       "      <td>WNT3</td>\n",
       "      <td>WNT7B</td>\n",
       "      <td>9606.ENSP00000225512</td>\n",
       "      <td>9606.ENSP00000341032</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.924</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.656</td>\n",
       "      <td>0.904</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8612</th>\n",
       "      <td>XKR4</td>\n",
       "      <td>XKRX</td>\n",
       "      <td>9606.ENSP00000328326</td>\n",
       "      <td>9606.ENSP00000362047</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.442</td>\n",
       "      <td>0.442</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8613</th>\n",
       "      <td>ZIC2</td>\n",
       "      <td>ZIC5</td>\n",
       "      <td>9606.ENSP00000365514</td>\n",
       "      <td>9606.ENSP00000267294</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.882</td>\n",
       "      <td>0.616</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.771</td>\n",
       "      <td>0.651</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8614 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #node1     node2       node1_string_id       node2_string_id  \\\n",
       "0     A2ML1     CWH43  9606.ENSP00000299698  9606.ENSP00000226432   \n",
       "1     A2ML1      DSG3  9606.ENSP00000299698  9606.ENSP00000257189   \n",
       "2     A2ML1     PADI2  9606.ENSP00000299698  9606.ENSP00000364635   \n",
       "3     A2ML1  SERPINB7  9606.ENSP00000299698  9606.ENSP00000381101   \n",
       "4     A2ML1      KLK6  9606.ENSP00000299698  9606.ENSP00000366047   \n",
       "...     ...       ...                   ...                   ...   \n",
       "8609   WNT2       WT1  9606.ENSP00000265441  9606.ENSP00000331327   \n",
       "8610   WNT2     WNT7B  9606.ENSP00000265441  9606.ENSP00000341032   \n",
       "8611   WNT3     WNT7B  9606.ENSP00000225512  9606.ENSP00000341032   \n",
       "8612   XKR4      XKRX  9606.ENSP00000328326  9606.ENSP00000362047   \n",
       "8613   ZIC2      ZIC5  9606.ENSP00000365514  9606.ENSP00000267294   \n",
       "\n",
       "      neighborhood_on_chromosome  gene_fusion  phylogenetic_cooccurrence  \\\n",
       "0                            0.0          0.0                        0.0   \n",
       "1                            0.0          0.0                        0.0   \n",
       "2                            0.0          0.0                        0.0   \n",
       "3                            0.0          0.0                        0.0   \n",
       "4                            0.0          0.0                        0.0   \n",
       "...                          ...          ...                        ...   \n",
       "8609                         0.0          0.0                        0.0   \n",
       "8610                         0.0          0.0                        0.0   \n",
       "8611                         0.0          0.0                        0.0   \n",
       "8612                         0.0          0.0                        0.0   \n",
       "8613                         0.0          0.0                        0.0   \n",
       "\n",
       "      homology  coexpression  experimentally_determined_interaction  \\\n",
       "0        0.000         0.000                                  0.000   \n",
       "1        0.000         0.098                                  0.000   \n",
       "2        0.000         0.000                                  0.000   \n",
       "3        0.000         0.000                                  0.000   \n",
       "4        0.000         0.000                                  0.050   \n",
       "...        ...           ...                                    ...   \n",
       "8609     0.000         0.000                                  0.084   \n",
       "8610     0.925         0.000                                  0.000   \n",
       "8611     0.924         0.000                                  0.000   \n",
       "8612     0.000         0.000                                  0.000   \n",
       "8613     0.882         0.616                                  0.000   \n",
       "\n",
       "      database_annotated  automated_textmining  combined_score  \n",
       "0                    0.0                 0.475           0.475  \n",
       "1                    0.0                 0.370           0.408  \n",
       "2                    0.0                 0.466           0.466  \n",
       "3                    0.0                 0.550           0.550  \n",
       "4                    0.0                 0.717           0.719  \n",
       "...                  ...                   ...             ...  \n",
       "8609                 0.0                 0.385           0.413  \n",
       "8610                 0.9                 0.668           0.904  \n",
       "8611                 0.9                 0.656           0.904  \n",
       "8612                 0.0                 0.442           0.442  \n",
       "8613                 0.0                 0.771           0.651  \n",
       "\n",
       "[8614 rows x 13 columns]"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_DEG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "id": "4059acae",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create an empty dataframe：\n",
    "def create_df_ini(arr_matrix_col):\n",
    "    form_header = ['']\n",
    "    for item in arr_matrix_col:\n",
    "        form_header.append(item)\n",
    "    df = pd.DataFrame(columns = form_header)\n",
    "    df[''] = arr_matrix_col\n",
    "    return df\n",
    "\n",
    "# Create the adjacent matrix：\n",
    "def create_matrix(arr_matrix_col,df_in_form,df_ini):\n",
    "    cnt = 0\n",
    "    arr = []\n",
    "    for item in df_in_form.columns:\n",
    "        arr.append(item)\n",
    "    ele0 = arr[0]\n",
    "    ele1 = arr[1]\n",
    "    if ele1 == 'lnc_DGE':\n",
    "        ele1 = arr[2]\n",
    "    \n",
    "    print(ele0,end=\" \")\n",
    "    print(ele1)\n",
    "    \n",
    "    x = 0\n",
    "    for elem_m in df_in_form[ele0]:\n",
    "        elem_l = df_in_form[ele1][cnt]\n",
    "        cnt = cnt + 1\n",
    "        \n",
    "        # flag: mark the position of the gene in the matrix(the column and row information)\n",
    "        if elem_l in arr_matrix_col and elem_m in arr_matrix_col:\n",
    "            flag_m = arr_matrix_col.index(elem_m)\n",
    "            flag_l = arr_matrix_col.index(elem_l)\n",
    "            \n",
    "            x = x + 1\n",
    "            df_ini.at[flag_m,elem_l] = 1\n",
    "            df_ini.at[flag_l,elem_m] = 1\n",
    "    \n",
    "    print(x)\n",
    "    return df_ini\n",
    "\n",
    "# Create a function to make diagonal elements in the matrix equals to 0：\n",
    "def set_zero(arr_matrix_col,df):\n",
    "    #set the diagonal equals to 0，that is gene can not reach itself：\n",
    "    for i in range(0,len(arr_matrix_col)):\n",
    "        temp = arr_matrix_col[i]\n",
    "        for j in range(0,len(df[temp])):\n",
    "            if i == j:\n",
    "                df.at[j,temp] = 0\n",
    "    #return the dataframe：\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "id": "9166a65b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "lncRNA gene\n",
      "431\n",
      "431.0\n",
      "miRNA target_symbol\n",
      "508\n",
      "938.0\n",
      "miR_DGE geneName\n",
      "1777\n",
      "1729.0\n",
      "#node1 node2\n",
      "8614\n",
      "10343.0\n"
     ]
    }
   ],
   "source": [
    "# Create the adjacent matrix：\n",
    "dataframe2 = create_df_ini(arr_matrix_columns)\n",
    "# input df1 to df3 to fill dataframe2, this step only fill 1 in the matrix:\n",
    "dataframe2 = create_matrix(arr_matrix_columns,df1,dataframe2)\n",
    "cnt = 0\n",
    "arr_no_one = []\n",
    "for elem in arr_matrix_columns:\n",
    "    list_cnt = []\n",
    "    list_cnt = dataframe2[elem]\n",
    "    for item in list_cnt:\n",
    "        if item == 1:\n",
    "            cnt = cnt+1\n",
    "print(cnt/2)\n",
    "dataframe2 = create_matrix(arr_matrix_columns,df2,dataframe2)\n",
    "cnt = 0\n",
    "arr_no_one = []\n",
    "for elem in arr_matrix_columns:\n",
    "    list_cnt = []\n",
    "    list_cnt = dataframe2[elem]\n",
    "    for item in list_cnt:\n",
    "        if item == 1:\n",
    "            cnt = cnt+1\n",
    "print(cnt/2)\n",
    "dataframe2 = create_matrix(arr_matrix_columns,df3,dataframe2)\n",
    "cnt = 0\n",
    "arr_no_one = []\n",
    "for elem in arr_matrix_columns:\n",
    "    list_cnt = []\n",
    "    list_cnt = dataframe2[elem]\n",
    "    for item in list_cnt:\n",
    "        if item == 1:\n",
    "            cnt = cnt+1\n",
    "print(cnt/2)\n",
    "dataframe2 = create_matrix(arr_matrix_columns,df_DEG,dataframe2)\n",
    "cnt = 0\n",
    "arr_no_one = []\n",
    "for elem in arr_matrix_columns:\n",
    "    list_cnt = []\n",
    "    list_cnt = dataframe2[elem]\n",
    "    for item in list_cnt:\n",
    "        if item == 1:\n",
    "            cnt = cnt+1\n",
    "print(cnt/2)\n",
    "\n",
    "# make the diagonal 0：\n",
    "dataframe2 = set_zero(arr_matrix_columns,dataframe2)\n",
    "#dataframe2.info()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "01461d0e",
   "metadata": {},
   "source": [
    "# Data scale = 1748×1748，Number of edges = 10343"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "id": "d51d3a19",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>hsa-let-7b-5p</th>\n",
       "      <th>hsa-let-7c-5p</th>\n",
       "      <th>hsa-let-7f-5p</th>\n",
       "      <th>hsa-miR-101-3p</th>\n",
       "      <th>hsa-miR-101-5p</th>\n",
       "      <th>hsa-miR-103a-3p</th>\n",
       "      <th>hsa-miR-106a-5p</th>\n",
       "      <th>hsa-miR-107</th>\n",
       "      <th>hsa-miR-1179</th>\n",
       "      <th>...</th>\n",
       "      <th>VSIG2</th>\n",
       "      <th>SLC17A9</th>\n",
       "      <th>PLEKHN1</th>\n",
       "      <th>SEC14L5</th>\n",
       "      <th>ZNF695</th>\n",
       "      <th>TMEM82</th>\n",
       "      <th>VIT</th>\n",
       "      <th>SLC5A12</th>\n",
       "      <th>TESC</th>\n",
       "      <th>TMPRSS5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hsa-let-7b-5p</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hsa-let-7c-5p</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hsa-let-7f-5p</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hsa-miR-101-3p</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hsa-miR-101-5p</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1743</th>\n",
       "      <td>TMEM82</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1744</th>\n",
       "      <td>VIT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1745</th>\n",
       "      <td>SLC5A12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1746</th>\n",
       "      <td>TESC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1747</th>\n",
       "      <td>TMPRSS5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1748 rows × 1749 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     hsa-let-7b-5p hsa-let-7c-5p hsa-let-7f-5p hsa-miR-101-3p  \\\n",
       "0      hsa-let-7b-5p             0           NaN           NaN            NaN   \n",
       "1      hsa-let-7c-5p           NaN             0           NaN            NaN   \n",
       "2      hsa-let-7f-5p           NaN           NaN             0            NaN   \n",
       "3     hsa-miR-101-3p           NaN           NaN           NaN              0   \n",
       "4     hsa-miR-101-5p           NaN           NaN           NaN            NaN   \n",
       "...              ...           ...           ...           ...            ...   \n",
       "1743          TMEM82           NaN           NaN           NaN            NaN   \n",
       "1744             VIT           NaN           NaN           NaN            NaN   \n",
       "1745         SLC5A12           NaN           NaN           NaN            NaN   \n",
       "1746            TESC           NaN           NaN           NaN            NaN   \n",
       "1747         TMPRSS5           NaN           NaN           NaN            NaN   \n",
       "\n",
       "     hsa-miR-101-5p hsa-miR-103a-3p hsa-miR-106a-5p hsa-miR-107 hsa-miR-1179  \\\n",
       "0               NaN             NaN             NaN         NaN          NaN   \n",
       "1               NaN             NaN             NaN         NaN          NaN   \n",
       "2               NaN             NaN             NaN         NaN          NaN   \n",
       "3               NaN             NaN             NaN         NaN          NaN   \n",
       "4                 0             NaN             NaN         NaN          NaN   \n",
       "...             ...             ...             ...         ...          ...   \n",
       "1743            NaN             NaN             NaN         NaN          NaN   \n",
       "1744            NaN             NaN             NaN         NaN          NaN   \n",
       "1745            NaN             NaN             NaN         NaN          NaN   \n",
       "1746            NaN             NaN             NaN         NaN          NaN   \n",
       "1747            NaN             NaN             NaN         NaN          NaN   \n",
       "\n",
       "      ... VSIG2 SLC17A9 PLEKHN1 SEC14L5 ZNF695 TMEM82  VIT SLC5A12 TESC  \\\n",
       "0     ...   NaN     NaN     NaN     NaN    NaN    NaN  NaN     NaN  NaN   \n",
       "1     ...   NaN     NaN     NaN     NaN    NaN    NaN  NaN     NaN  NaN   \n",
       "2     ...   NaN     NaN     NaN     NaN    NaN    NaN  NaN     NaN  NaN   \n",
       "3     ...   NaN     NaN     NaN     NaN    NaN    NaN  NaN     NaN  NaN   \n",
       "4     ...   NaN     NaN     NaN     NaN    NaN    NaN  NaN     NaN  NaN   \n",
       "...   ...   ...     ...     ...     ...    ...    ...  ...     ...  ...   \n",
       "1743  ...   NaN     NaN     NaN     NaN    NaN      0  NaN     NaN  NaN   \n",
       "1744  ...   NaN     NaN     NaN     NaN    NaN    NaN    0     NaN  NaN   \n",
       "1745  ...   NaN     NaN     NaN     NaN    NaN    NaN  NaN       0  NaN   \n",
       "1746  ...   NaN     NaN     NaN     NaN    NaN    NaN  NaN     NaN    0   \n",
       "1747  ...   NaN     NaN     NaN     NaN    NaN    NaN  NaN     NaN  NaN   \n",
       "\n",
       "     TMPRSS5  \n",
       "0        NaN  \n",
       "1        NaN  \n",
       "2        NaN  \n",
       "3        NaN  \n",
       "4        NaN  \n",
       "...      ...  \n",
       "1743     NaN  \n",
       "1744     NaN  \n",
       "1745     NaN  \n",
       "1746     NaN  \n",
       "1747       0  \n",
       "\n",
       "[1748 rows x 1749 columns]"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataframe2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "3c6166f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# create a function to make the dataframe into a .csv file:\n",
    "def dataframe_to_csv_file(FileName,df_need_to_trans):\n",
    "    df_need_to_trans.to_csv(FileName,index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "6cf12345",
   "metadata": {},
   "outputs": [],
   "source": [
    "dataframe_to_csv_file(r'string_data/D_adjacency_matrix.csv',dataframe2)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
